Document Cloud Functions gen2 and Cloud Events support (#23717)
quarkusio/quarkus: docs/src/main/asciidoc/funqy-gcp-functions.adoc

This will compile and package your code.

== Creating the functions

In this example, we will create two background functions and a cloud events function.

> Background functions allow to react to Google Cloud events like PubSub messages, Cloud Storage events, Firestore events, ...

Reviewer: ```suggestion
Background functions allow you to react to Google Cloud events like PubSub messages, Cloud Storage events, Firestore events, ...
```

> Cloud events functions allow to react to supported events using the Cloud Events specification.

Reviewer: ```suggestion
Cloud events functions allow you to react to supported events using the Cloud Events specification.
```

this is a convenient class to deserialize a PubSub message.
3. This is a background function that takes as parameter a `io.quarkus.funqy.gcp.functions.event.StorageEvent`,
this is a convenient class to deserialize a Google Storage event.
4. This is a cloud events functions, that takes as parameter a `io.cloudevents.CloudEvent`,

> inside it the `getData()` method will return the event content, here the storage event.

Reviewer: ```suggestion
inside it the `getData()` method will return the event content, storage event in this case.
```
